examples/esbuild/src/BUILD.bazel

```diff
@@ -0,0 +1,53 @@
+load("@build_bazel_rules_nodejs//:index.bzl", "generated_file_test", "nodejs_binary", "npm_package_bin")
+load("@npm//@bazel/esbuild:index.bzl", "esbuild")
+load("@npm//@bazel/typescript:index.bzl", "ts_project")
+
+# build the Typescript to JS with ts_project
+ts_project(
+    name = "lib",
+    srcs = [
+        "main.ts",
+        "name.ts",
+    ],
+    tsconfig = {},
+)
+
+# create a single bundle of the JS sources
+esbuild(
+    name = "bundle",
+    entry_point = "main.ts",
+    minify = True,
+    # setting node as the platform will default us to CJS
+    platform = "node",
+    tool = select({
+        "@bazel_tools//src/conditions:darwin": "@esbuild_darwin//:bin/esbuild",
```

Comment:
this is fine to be consistent with other places but I think we are supposed to use `@platforms` in the future https://github.com/bazelbuild/bazel/issues/8622

Reply:
Made https://github.com/bazelbuild/rules_nodejs/issues/2440 to track these